compiler/rustc_codegen_ssa/src/mir/place.rs

```diff
@@ -154,14 +153,6 @@ impl<'a, 'tcx, V: CodegenObject> PlaceRef<'tcx, V> {
             }
             _ if field.is_sized() => return simple(),
             ty::Slice(..) | ty::Str | ty::Foreign(..) => return simple(),
-            ty::Adt(def, _) => {
-                if def.repr().packed() {
```

Comment:
We could have tried to preserve this fast-path for some packed structs, but I don't think that's worth it.